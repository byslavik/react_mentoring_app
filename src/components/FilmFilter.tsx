import * as React from 'react';

import { connect } from 'react-redux';


import * as s from './scss/StatusBar.scss';

class FilmFilter extends React.Component<any, any> {
  
  sortFilms = (byfield:string)=> {
    this.props.sortFilms(byfield);
  }

    render() {
        return (<ul>
          <li style={{fontWeight: "bold"}}>Sort By: </li>
          <li><button className={this.props.sort == 'vote_count' ? s.active : ''} onClick={this.sortFilms.bind(this, 'vote_count')}>by Votes Count</button></li>
          <li><button className={this.props.sort == 'popularity' ? s.active : ''} onClick={this.sortFilms.bind(this, 'popularity')}>by Popularity</button></li>
      </ul> )
    }
}

export default connect(
  (state: any) => ({
    sort: state.sort
  }),
  (dispatch: any) => ({
    sortFilms: (byfield:string)=> {
      dispatch({type: "SORT_FILMS", payload: byfield});
    }
   })
)(FilmFilter);