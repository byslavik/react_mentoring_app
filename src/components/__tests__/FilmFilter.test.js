import * as React from 'react';
import {FilmFilter} from '../FilmFilter';
import * as renderer from 'react-test-renderer';

describe('FilmFilter component', ()=>{
  it('FilmFilter component shows correctly with "vote_count" option', () => {
    const tree = renderer.create(
      <FilmFilter sort="vote_count"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('FilmFilter component shows correctly with "popularity" option', () => {
    const tree = renderer.create(
      <FilmFilter sort="popularity"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});