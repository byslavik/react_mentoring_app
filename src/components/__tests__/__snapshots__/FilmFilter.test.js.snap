// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmFilter component FilmFilter component shows correctly with "popularity" option 1`] = `
<ul>
  <li
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Sort By: 
  </li>
  <li>
    <button
      className=""
      onClick={[Function]}
    >
      by Votes Count
    </button>
  </li>
  <li>
    <button
      className={undefined}
      onClick={[Function]}
    >
      by Popularity
    </button>
  </li>
</ul>
`;

exports[`FilmFilter component FilmFilter component shows correctly with "vote_count" option 1`] = `
<ul>
  <li
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Sort By: 
  </li>
  <li>
    <button
      className={undefined}
      onClick={[Function]}
    >
      by Votes Count
    </button>
  </li>
  <li>
    <button
      className=""
      onClick={[Function]}
    >
      by Popularity
    </button>
  </li>
</ul>
`;
